closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -145,6 +191,15 @@ void init(int opmode, Key key, GCMParameterSpec spec)
             }
             this.key = keyValue.clone();
 
+            /*
+             * Check whether cipher and IV need to be set,
+             * whether because something changed here or
+             * a call to set them in context hasn't been
+             * made yet.
+             */
+            newIVLen = (lastIv.length != this.iv.length) || newIVLen;
+            newKeyLen = (lastKey.length != this.key.length) || newKeyLen;
```

Comment:
I think this should be expressed more simply:
```java
            if (lastIv.length != this.iv.length) {
                newIVLen = true;
            }
            if (lastKey.length != this.key.length) {
                newKeyLen = true;
            }
```

Reply:
I can change it to this form, but I still need to check its previous value. If two consecutive calls to `init()` are made, they will be set to `false`, but since no call to `doFinal()` will have been made the cipher will never be initialized and added to the context.

Reply:
Please elaborate. There are four cases to consider: the field was true or false before and the lengths are equal or not; in all cases the two code forms have the same result. I don't understand your concern.

Reply:
Changed.